detection_rules/rule.py

```diff
@@ -1145,10 +1147,12 @@ def post_conversion_validation(self, value: dict, **kwargs):
         data: AnyRuleData = value["data"]
         metadata: RuleMeta = value["metadata"]
 
-        data.validate_query(metadata)
-        data.data_validator.validate_note()
-        data.data_validator.validate_bbr(metadata.get('bypass_bbr_timing'))
-        data.validate(metadata) if hasattr(data, 'validate') else False
+        test_config = RULES_CONFIG.test_config
+        if not test_config.check_skip_by_rule_id(value['data'].rule_id):
+            data.validate_query(metadata)
```

Comment:
I see here that users can bypass validation. I'm thinking about different examples where this may break down.

What happens if a user wants to validate their query but use non-ecs fields and dataviews instead of specifying index or dataset information within the query?

Reply:
either
1. BYOS
2. non-ecs-schema
3. bypass query validation and then write a custom validator method (just parsing or whatever)

Reply:
would be there own, but as part of BYOS updates, that support would need to be added